### ISSUE TITLE
Comparison between signed and unsigned integer warning, results in error.

### ISSUE THREAD

**Issue by kenOfYugen** [frewsxcv/afl.rs#98]
While following the steps in [the tutorial](https://frewsxcv.github.io/afl.rs/tutorial.html), when attempting `cargo build` I get the following:

```
$ cargo build
   Compiling afl v0.1.4
   Compiling afl-sys v0.1.4
   Compiling afl-plugin v0.1.4
error: failed to run custom build command for `afl-plugin v0.1.4`
process didn't exit successfully: `/home/ken/Desktop/afl-test/rust-url/target/debug/build/afl-plugin-62cbc3ca789f3664/build-script-build` (exit code: 101)
--- stdout
TARGET = Some("i686-unknown-linux-gnu")
PROFILE = Some("debug")
TARGET = Some("i686-unknown-linux-gnu")
debug=true opt-level=2
HOST = Some("i686-unknown-linux-gnu")
TARGET = Some("i686-unknown-linux-gnu")
TARGET = Some("i686-unknown-linux-gnu")
HOST = Some("i686-unknown-linux-gnu")
CXX_i686-unknown-linux-gnu = None
CXX_i686_unknown_linux_gnu = None
HOST_CXX = None
CXX = None
HOST = Some("i686-unknown-linux-gnu")
TARGET = Some("i686-unknown-linux-gnu")
HOST = Some("i686-unknown-linux-gnu")
CXXFLAGS_i686-unknown-linux-gnu = None
CXXFLAGS_i686_unknown_linux_gnu = None
HOST_CXXFLAGS = None
CXXFLAGS = None
running: "c++" "-O2" "-ffunction-sections" "-fdata-sections" "-g" "-m32" "-march=i686" "-fPIC" "-Wall" "-Werror" "-fno-rtti" "-c" "-I/usr/include" "-march=i686" "-mtune=generic" "-O2" "-pipe" "-fstack-protector-strong" "-fPIC" "-fvisibility-inlines-hidden" "-Wall" "-W" "-Wno-unused-parameter" "-Wwrite-strings" "-Wcast-qual" "-Wno-missing-field-initializers" "-pedantic" "-Wno-long-long" "-Wno-maybe-uninitialized" "-Wdelete-non-virtual-dtor" "-Wno-comment" "-std=c++11" "-ffunction-sections" "-fdata-sections" "-O3" "-DNDEBUG" "-fno-exceptions" "-D_GNU_SOURCE" "-D__STDC_CONSTANT_MACROS" "-D__STDC_FORMAT_MACROS" "-D__STDC_LIMIT_MACROS" "-o" "/home/ken/Desktop/afl-test/rust-url/target/debug/build/afl-plugin-62cbc3ca789f3664/out/afl-llvm-pass.so.o" "-c" "afl-llvm-pass.so.cc"
cargo:warning=afl-llvm-pass.so.cc: In member function ‘virtual bool {anonymous}::AFLCoverage::runOnModule(llvm::Module&)’:
cargo:warning=afl-llvm-pass.so.cc:113:28: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
cargo:warning=       if ((random() % 100) >= inst_ratio) continue;
cargo:warning=           ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
cargo:warning=cc1plus: all warnings being treated as errors
ExitStatus(ExitStatus(256))


command did not execute successfully, got: exit code: 1



--- stderr
thread 'main' panicked at 'explicit panic', /home/ken/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.37/src/lib.rs:952
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

While it could be possible to override `-Werror=sign-compare`, isn't it better to fix the comparison in the source file?
Please correct me me if I am wrong here.

---

EDIT: 
Just found out about #37. Isn't the fix included in `0.1.4`?

**Comment by frewsxcv** [comment]
The LLVM specific code in this repository is almost entirely from upstream AFL.

Here's the relevant upstream code:

https://github.com/mcarpenter/afl/blob/be3e88d639da5350603f6c0fee06970128504342/llvm_mode/afl-llvm-pass.so.cc#L115

https://github.com/mcarpenter/afl/blob/be3e88d639da5350603f6c0fee06970128504342/types.h#L71

And here's the relevant line in this repository:

https://github.com/frewsxcv/afl.rs/blob/8c52dca097d78ed29956918d049d86e65ff3f0f5/afl-plugin/afl-llvm-pass.so.cc#L113

Which C++ compiler is being used on your system and which version? Also, can you still replicate the issue?

**Comment by kenOfYugen** [comment]
>  Which C++ compiler is being used on your system and which version?

From `$ c++ -v` => `gcc version 6.2.1 20160830 (GCC) `

> Also, can you still replicate the issue?

Yes, I just tried a fresh setup and `cargo b` still reports the same comparison error before halting (`v0.1.5`).

I saw that the tutorial was updated to use docker.
Unfortunately I can not try it via this approach on this specific machine, since it's a 32bit system which I believe docker doesn't support. If compilation via docker is a permanent solution, then maybe this issue should be closed.

I can confirm that [casting the unsigned integer to an integer](https://github.com/kenOfYugen/afl.rs/blob/master/afl-plugin/afl-llvm-pass.so.cc#L113) solves the issue.
The bad news is that even after `afl-llvm-pass.so.cc` builds, #57 happens.
